flang/lib/Lower/Bridge.cpp

```diff
@@ -1852,38 +1848,46 @@ class FirConverter : public Fortran::lower::AbstractConverter {
       instantiateLocal(var);
   }
 
+  void mapDummyAndResults(const Fortran::lower::CalleeInterface &callee) {
+    assert(builder && "need a builder at this point");
+    using PassBy = Fortran::lower::CalleeInterface::PassEntityBy;
+    auto mapPassedEntity = [&](const auto arg) -> void {
+      switch (arg.passBy) {
+      case PassBy::BaseAddress:
+        LLVM_FALLTHROUGH;
+      case PassBy::BoxChar:
+        LLVM_FALLTHROUGH;
+      case PassBy::Descriptor:
+        LLVM_FALLTHROUGH;
+      case PassBy::Value:
+        addSymbol(arg.entity.get(), arg.firArgument);
+        break;
+      case PassBy::AddressAndLength:
```

Comment:
This would be less code with a simple `if` and `else`.

Reply:
Good point, done.